### PR TITLE
Feat/ft out exp proper test

### PR DESCRIPTION
**Pull request by kirilklein**
## Summary by CodeRabbit

- **New Features**
  - Added GitHub Actions workflows for advanced cohort selection and causal pipeline performance testing.
  - Introduced causal inference dataset classes supporting exposure and outcome labels.
  - Developed joint exposure-outcome prediction models with specialized neural network heads.
  - Added fine-tuning modules with cross-validation, calibration, and performance aggregation.
  - Created new configuration files for causal finetuning, cohort selection, calibration, and data preparation.
  - Added scripts for preparing fine-tuning data, simulating causal effects, and validating calibration and estimates.
  - Introduced diagnostic and calibration plotting utilities for causal inference predictions.
  - Added performance testing workflows and validation scripts for fold consistency and estimate accuracy.
  - Developed causal simulation utilities and data shard management for generating causal-effect datasets.
  - Expanded example datasets with causal effect annotations and documentation.
  - Added unit and integration tests for causal models, data utilities, and calibration workflows.

- **Bug Fixes**
  - Corrected import paths for causal modules and classes.
  - Enhanced path validation and environment setup in workflows and scripts.

- **Chores**
  - Refactored CI workflows for modularity and clarity.
  - Simplified Windows batch scripts for causal pipeline execution.
  - Removed deprecated calibration and training helper modules.
  - Updated dependencies to include matplotlib for plotting.

- **Documentation**
  - Added README for causal data simulation process.
  - Improved inline comments and docstrings in new causal modules.

- **Tests**
  - Added tests for calibration quality, fold consistency, and performance metrics.
  - Included unit tests for causal model heads and dataset classes.
  - Added validation scripts for data integrity and estimate correctness.

### CODE REVIEW

File: corebehrt/configs/causal/finetune/prepare/ft_exp_uncensored.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+data:
+  type: "finetune"
+  truncation_len: 64
+  min_len: 1 # 0 by default
+
+outcome: # we will convert outcomes to binary based on whether at least one outcome is in the follow up window
+  n_hours_censoring: 10 # censor time after index date (negative means before)
+  n_hours_start_follow_up: -10 # start follow up (considering outcomes) time after index date (negative means before)
+  n_hours_end_follow_up: null # end follow up (considering outcomes) time after index date (negative means before)
```
[the remark this answers is not in the document]
Comment (kirilklein) [reply corebehrt/configs/causal/finetune/prepare/ft_exp_uncensored.yaml:24]: It's actually correct, because we are censoring data 10 hours after the event occurs, so it is in our input data, and thus uncensored
